### PR TITLE
Various fixes, add binary listing & branching

### PR DESCRIPTION
**Pull request by refi64**
Did *not* realize this PR was that big until I, err, opened it. Includes #3.

### CODE REVIEW

File: open-build-service-api/src/lib.rs [hunk 1]
```diff
@@ -48,16 +49,18 @@ impl std::fmt::Display for ApiError {
...
 pub enum RebuildMode {
     Transitive,
     Direct,
     Local,
 }
 
-#[derive(Clone, Copy, Deserialize, Debug, Eq, PartialEq)]
+#[derive(Clone, Copy, Deserialize, Debug, Eq, PartialEq, Display)]
 #[serde(rename_all = "lowercase")]
+#[strum(serialize_all = "snake_case")]
```
Comment (sjoerdsimons) [inline open-build-service-api/src/lib.rs:63]: serde lowercase  but strum snake_case ? (thought it won't matter, it's still odd)

File: open-build-service-api/src/lib.rs [hunk 2]
```diff
@@ -584,7 +589,11 @@ impl<'a> PackageBuilder<'a> {
         Ok(())
     }
 
-    pub async fn commit(&self, filelist: &CommitFileList) -> Result<CommitResult> {
+    pub async fn commit(
+        &self,
+        filelist: &CommitFileList,
+        options: &CommitOptions,
```
[the remark this answers is not in the document]
Comment (refi64) [reply open-build-service-api/src/lib.rs:736]: Yup, in particular once our OBS is upgraded, `withvalidate` would be ideal to specify.

File: open-build-service-mock/src/api/source.rs [hunk 3]
```diff
@@ -513,12 +524,113 @@ impl Respond for PackageSourceCommandResponder {
...
+                        Cow::Owned(format!(
+                            "home:{}:branches:{}",
+                            self.mock.auth().username(),
+                            origin_project_name
+                        ))
+                    });
+                let target_package_name = find_query_param(request, "target_package")
+                    .unwrap_or(Cow::Borrowed(origin_package_name));
+                let force = find_query_param(request, "force").is_some();
+                let missingok = find_query_param(request, "missingok").is_some();
```
[the remark this answers is not in the document]
Comment (refi64) [reply open-build-service-mock/src/api/source.rs:612]: I tested it out, pretty much any value here works, *even the empty string*.

File: open-build-service-api/src/lib.rs [hunk 4]
```diff
@@ -444,16 +457,23 @@ pub struct PackageBuilder<'a> {
 }
 
 impl<'a> PackageBuilder<'a> {
-    fn full_request(&self, repository: &str, arch: &str, command: &str) -> Result<Url> {
+    fn full_request(&self, repository: &str, arch: &str, command: Option<&str>) -> Result<Url> {
```
[the remark this answers is not in the document]
Comment (refi64) [reply open-build-service-api/src/lib.rs:561]: I'm admittedly not sure what the advantage would be of a custom enum here, unless I'm misunderstanding?

**Pull request merged**